packages/commonwealth/client/scripts/views/pages/CommunityManagement/Topics/CreateTopicsSection/CreateTopicSection.tsx

```diff
@@ -91,45 +86,41 @@ export const CreateTopicSection = () => {
         'char',
       )}
     ${disallowedCharMatches.join(', ')} are not permitted`;
-      setErrorMsg(err);
+      setNameErrorMsg(err);
       return ['failure', err];
     }
 
-    if (errorMsg) {
-      setErrorMsg(null);
-    }
+    setNameErrorMsg(null);
 
     return ['success', 'Valid topic name'];
   };
 
+  // TODO: implement logic around creating a new thread with new topic
   return (
     <div className="CreateTopicSection">
-      <div className="form">
+      <CWForm
+        validationSchema={topicCreationValidationSchema}
+        onSubmit={handleCreateTopic}
+      >
         <div className="form-inputs">
-          <div>
-            <CWTextInput
-              label="Name (required)"
-              placeholder="Enter a topic name"
-              value={name}
-              onInput={(e) => {
-                setName(e.target.value);
-              }}
-              inputValidationFn={handleInputValidation}
-              autoFocus
-            />
-            {errorMsg && (
-              <CWValidationText
-                className="validation-text"
-                message={errorMsg}
-                status="failure"
-              />
-            )}
-          </div>
-
           <CWTextInput
+            hookToForm
+            label="Name (required)"
+            placeholder="Enter a topic name"
+            name="topicName"
+            value={name}
+            onInput={(e) => {
+              setName(e.target.value);
+              handleInputValidation(e.target.value);
+            }}
+            customError={nameErrorMsg}
+            autoFocus
+          />
+          <CWTextInput
+            hookToForm
             label="Description"
             placeholder="Enter a description"
-            name="description"
+            name="topicDescription"
             tabIndex={2}
             value={description}
```

Comment:
we can remove the `value` and `onInput` props and its state `[description, setDescription]`. The `hookToForm` prop covers field updates

Reply:
Removed!